client/src/components/ratingsReviews/RatingsAndReviews.jsx

```diff
@@ -217,21 +220,59 @@ class RatingsAndReviews extends React.Component {
     });
   }
 
+  updateRatingsList(filteredList) {
+    this.setState({
+      filteredRatings: filteredList
```

Comment:
There's a lot to be accounted for if you're going to assign an array or object to component state. By default, React determines whether to update based on reference, not content. If you pass a new array/object it will automatically trigger a re-render, even if the content is the same, if you're trying to update state with changes to an existing array/object, nothing will re-render. If you need to use an array/object here, and you don't necessarily want to re-render identical content, write a custom `shouldComponentUpdate` method that does a more detailed comparison.

Reply:
I added your comment to a ticket and will work on writing a custom shouldComponentUpdate